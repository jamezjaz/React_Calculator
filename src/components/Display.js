import React from 'react';
import PropTypes from 'prop-types';
import '../styles/Display.css';

const Display = props => {
  const { result } = props;
  return (
    <div className="display">
      <p style={{ fontSize: '2rem', textAlign: 'end' }}>
        {result}
      </p>
    </div>
  );
};

Display.propTypes = {
  result: PropTypes.string,
};

Display.defaultProps = {
  result: 0,
};

export default Display;
