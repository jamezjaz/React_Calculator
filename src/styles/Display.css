.display {
  display: flex;
  width: 700px;
  background-color: #858693;
  height: 100px;
  color: #fff;
  font-weight: bold;
  padding: 7px;
  justify-content: flex-end;
}
