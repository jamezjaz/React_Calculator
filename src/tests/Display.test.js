import React from 'react';
import renderer from 'react-test-renderer';
import Display from '../components/Display';

it('renders correctly', () => {
  const tree = renderer
    .create(<Display result='0' />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});