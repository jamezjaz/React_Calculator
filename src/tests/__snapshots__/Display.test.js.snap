// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Display /> renders correctly 1`] = `
<div
  className="display"
>
  <p
    style={
      Object {
        "fontSize": "2rem",
        "textAlign": "end",
      }
    }
  >
    0
  </p>
</div>
`;
